### ISSUE TITLE
Report platform specific fragments to c8y

### ISSUE THREAD

**Issue by itsyitsy** [thin-edge/thin-edge.io#686]
As a user of thin-edge I would like to add some optional information about my device to the cloud representation.

The idea for this feature is to have the content(as in example) in a predefined path in `"/etc/tedge"`. The mapper should pick up this file and upload it to c8y.

It was considered to have that this configs in a cloud independent, in this case we would have to limit what fields we are supporting(that are mapped to the cloud platforms).

An example:
```json
{
  "c8y_RequiredAvailability": {
      "responseInterval": 5
  },
  "c8y_Firmware": {
      "name": "raspberrypi-bootloader",
      "version": "1.20140107-1",
      "url": "31aab9856861b1a587e2094690c2f6e272712cb1"
  },
  "c8y_Hardware": {
      "model": "BCM2708",
      "revision": "000e",
      "serialNumber": "00000000e2f5ad4d"
  }
}
```

You can also refer to [https://github.com/SoftwareAG/cumulocity-self-service-certification/blob/main/docs/vendor-requirements.md#hardware-information](https://github.com/SoftwareAG/cumulocity-self-service-certification/blob/main/docs/vendor-requirements.md#hardware-information)

AC:
- Introduce config file for this additional fragments
- Make Sure this fragments are reported to c8y
- Default should (at least) contain c8y_Agent fragment with the version of thin-edge.
```json
"c8y_Agent": {
    "name": "thin-edge.io",
    "version": "x.x.x"
}
```
- Do not write c8y_Agent fragment to file, just attach it to the file.

**Comment by itsyitsy** [comment]
Related to [https://github.com/thin-edge/thin-edge.io/issues/684](https://github.com/thin-edge/thin-edge.io/issues/684)

**Comment by itsyitsy** [comment]
This is the start for fragments support and will lateron be extended (Hooking mechanism and other ideas)

**Comment by initard** [comment]
Extra: adding url field to agent fragment as it is also a mandatory field (implemented in #703 )
